console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,26 @@ type ConsolePluginSpec struct {
 	// i18n is the configuration of plugin's localization resources.
 	// +optional
 	I18n ConsolePluginI18n `json:"i18n"`
+	// csp is a list of Content Security Policy directives for the plugin.
+	CSP []ConsolePluginCSP `json:"csp"`
+}
+
+// DirectiveType is an enumeration of supported CSP directives
+type DirectiveType string
+
+const (
+	DefaultSrc DirectiveType = "default-src"
+	ScriptSrc  DirectiveType = "script-src"
+	ImgSrc     DirectiveType = "img-src"
+	StyleSrc   DirectiveType = "style-src"
+	FontSrc    DirectiveType = "font-src"
```

Comment:
We should describe what these are or link to documentation. A link to the MDN topic would be good (unless we have a policy against linking to 3rd party docs).

https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP

Reply:
From what I see we are using links for 3rd party docs in the API - https://github.com/openshift/api/blob/master/config/v1/types_dns.go#L79-L81